ql/src/java/org/apache/hadoop/hive/ql/exec/FileSinkOperator.java

```diff
@@ -1823,6 +1831,22 @@ private int getBucketProperty(Object row) {
     }
   }
 
+  /**
+   * Reuired for rebalancing compaction. Encodes the raw bucket property set by the compactor
+   * @param row The acid row in which the bucket needs to be updated.
+   */
+  private void setBucketProperty(Configuration hiveConf, Object row, int bucketId) {
+    //TODO: statementId?
+    BucketCodec codec = conf.getBucketingVersion() == 2 ? BucketCodec.V1 : BucketCodec.V0;
```

Comment:
haven't seen in a code BucketCodec.V0 being used, do we need this?

Reply:
Not sure to be honest. I just wanted to make sure the code covers all possible paths. Having it there should not cause issues if only V1 is used for ACID, but the opposite case would lead to errors.

Reply:
check VectorizedOrcAcidRowBatchReader.computeOffsetAndBucket(), we are not using V0 format anymore, we should still read those, but not write.
```
    int bucketId = AcidUtils.parseBucketId(file.getPath());
    int bucketProperty = BucketCodec.V1.encode(new AcidOutputFormat.Options(conf).bucket(bucketId))
````